opm/material/common/FastSmallVector.hpp

```diff
@@ -137,10 +139,46 @@ class FastSmallVector
     const ValueType& operator[](size_t idx) const
     { return dataPtr_[idx]; }
 
-    //! number of the element
-    size_t size() const
+    using size_type = typename std::vector<ValueType>::size_type;
+
+    //! number of elements
+    size_type size() const
     { return size_; }
 
+    //! Iterator type is a plain pointer, so be warned there is no validity checking.
+    using Iterator = ValueType*;
+
+    //! To support range-for etc.
+    Iterator begin() const
+    { return dataPtr_; }
+
+    //! To support range-for etc.
+    Iterator end() const
+    { return dataPtr_ + size_; }
+
+    size_type capacity()
+    { return size_ < N ? N : data_.capacity(); }
+
+    void push_back(const ValueType& value)
+    {
+        if (size_ < N) {
+            // Data is contained in smallBuf_
+            smallBuf_[size_++] = value;
+        } else if (size_ == N) {
+            // Must switch from using smallBuf_ to using data_
+            data_.resize(N + 1);
+            std::copy_n(smallBuf_.begin(), N, data_.begin());
+            data_[N] = value;
```

Comment:
Another option here would be to write this as
```C++
data_.reserve(N + 1);
data_.assign(smallBuf_.begin(), smallBuf_.end());
data_.push_back(value);
```
but to be honest I don't know which is faster, especially not without measuring.  You do avoid default-constructing (or direct-initialising) `ValueType` objects in the `resize()` call, though.

Reply:
I like this better, switching to this approach.